### PR TITLE
unit testing homework

### PR DESCRIPTION
**Pull request by NatalyaOvsyannikova**

### CODE REVIEW

File: src/test/java/com/rntgroup/tat/unittest/CalculatorTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        System.out.println("Before class");
+    }
+
+    @AfterClass
+    public void afterClass() {
+        System.out.println("After class");
+    }
+
+    @BeforeGroups(value = {"trigonometry"})
+    public void BeforeGroup() {
```
Comment (doctordrue) [inline src/test/java/com/rntgroup/tat/unittest/CalculatorTest.java]:
```suggestion
    public void beforeGroup() {
```

File: src/test/java/com/rntgroup/tat/unittest/CalculatorTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.rntgroup.tat.unittest;
+
+import com.epam.tat.module4.Calculator;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeGroups;
+
+public class CalculatorTest {
```
Comment (doctordrue) [inline src/test/java/com/rntgroup/tat/unittest/CalculatorTest.java]: make this class abstract and name like `BaseCalculatorTest`

File: src/test/java/com/rntgroup/tat/unittest/MultiplicationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public void testMult(double a, double b, double expected) {
+        double mult = calculator.mult(a, b);
+        Assert.assertEquals(mult, expected);
+    }
+
+    @DataProvider(name = "multDataProvider")
+    public static Object[][] multDataProvider() {
+        return new Object[][]{
+                {42, 1.5, 63},
+                {-15, 2, -30}
```
Comment (doctordrue) [inline src/test/java/com/rntgroup/tat/unittest/MultiplicationTest.java]: what about multiplying 2 negatives and using zero?

File: src/test/java/com/rntgroup/tat/unittest/SubtractionTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public void testSub(long a, long b, long expected) {
+        long sub = calculator.sub(a, b);
+        Assert.assertEquals(sub, expected);
+    }
+
+    @DataProvider(name = "subDataProvider")
+    public static Object[][] subDataProvider() {
+        return new Object[][]{
+                {1, 1, 0},
+                {5, 10, -5}
```
Comment (doctordrue) [inline src/test/java/com/rntgroup/tat/unittest/SubtractionTest.java]: add test to subtract negative value

File: src/test/java/com/rntgroup/tat/unittest/SumTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    public void testSum(long a, long b, long expected) {
+        long sum = calculator.sum(a, b);
+        Assert.assertEquals(sum, expected);
+    }
+
+    @DataProvider(name = "sumDataProvider")
+    public static Object[][] sumDataProvider() {
+        return new Object[][]{
+                {1, 1, 2},
+                {0, 0, 0}
```
Comment (doctordrue) [inline src/test/java/com/rntgroup/tat/unittest/SumTest.java]: sum with a negative value

**Review by doctordrue — CHANGES_REQUESTED** [review]
